database/db.go

```diff
@@ -110,16 +113,39 @@ func (db *DB) GetDocument(name string) (Document, error) {
 }
 
 func (db *DB) PutUser(user User) error {
-	_, err := db.conn.Exec(`INSERT INTO users (uuid) VALUES ($1) ON CONFLICT DO NOTHING`, user.UUID)
+	_, err := db.GetUser(user.UUID)
+	if err == ErrUserNotFound {
```

Comment:
I, personally, don't like the behaviour of creating a user by PUTting. In general, POST is used to create a source when working in a RESTful API. A PUT to a non-existent resource should give a 404 (since the resource doesn't exist)

Reply:
You are correct

A datapoint is that the implemented behaviour in this PR is consistent with what exists currently: https://github.com/alphagov/paas-accounts#put-documentsname